### ISSUE TITLE
[Research] Front-end solutions for AF pages and document next steps

### ISSUE THREAD

**Issue by JonellaCulmer** [fecgov/fec-cms#4552]
**What we're after:**
We have several improvements that need to be made to the AF pages. First we need to research and determine if these changes can be made. We need to document the feasibility and research and make recommendations on what can be accomplished now (and how) and what can't be done now (and why not).

**Action items/considerations:**
- Can we accurately and consistently replace the acronym RTB with Reason to believe?
- Can we accurately and consistently replace the penalty paid flag of Yes/No/Pending with paid/not yet paid/partially paid?
- Document any API necessary changes

### Related issues
- https://github.com/fecgov/fec-cms/issues/4446 Reach out to AF team to kickstart migration

### Completion criteria
- [ ] Document research and findings about the feasibility of the aforementioned considerations, including:
  - Consistently replace acronym "RTB" and display as "Reason to believe"
  - Replace penalty paid flag of Yes/No/Pending with paid/not paid/partially paid
- [ ] Open followup implementation tickets, as appropriate

**Comment by patphongs** [comment]
@JonellaCulmer This is what it looks like with the move of the disposition to it's own section. Does this look ok?

![Screen Shot 2021-05-04 at 4 39 02 PM](https://user-images.githubusercontent.com/12799132/117066853-45a0d580-acf7-11eb-8379-6036fa9ed024.png)

**Comment by JonellaCulmer** [comment]
@patphongs Thanks for the screenshot. We don't need the dividing lines per section. I also found the dividing line on the MURs page and that should go too. Here's a good example from AOs: https://www.fec.gov/data/legal/advisory-opinions/2021-04/.

**Comment by patphongs** [comment]
Acronym "RTB" has been replaced with "reason to believe" in the template. Awaiting civil penalty due date and payment status to be populated before showing that data.

**Comment by patphongs** [comment]
@JonellaCulmer This is an updated look as to what I have for the AF canonical page. As mentioned before, we need to find out what needs to be displayed when there is no value present. Currently we use either None or No amount. Can you review what we want to have in these situations?

![Screen Shot 2021-05-05 at 12 13 34 PM](https://user-images.githubusercontent.com/12799132/117174828-227c3180-ad9c-11eb-92a8-da0360dab3fb.png)

**Comment by JonellaCulmer** [comment]
@patphongs Need to reconfirm with Rhiannon as to whether or not it's a given that if something is referred to the treasury there will always be a dollar amount and a date. If there's a referral date but no dollar amount, we should say No amount. But if the referral date and referral amount are mutually exclusive I think we can eliminate that line entirely and we won't need language there.

**Comment by patphongs** [comment]
> But if the referral date and referral amount are mutually exclusive I think we can eliminate that line entirely and we won't need language there.

@JonellaCulmer How do you know if the referral date and referral amount are mutually exclusive in order to eliminate that row entirely?

**Comment by patphongs** [comment]
Similarly @JonellaCulmer, if there's no challenge received date, then can we eliminate that row as well?

**Comment by patphongs** [comment]
@JonellaCulmer I also just found that sometimes there's no text for the challenge outcome even though there's a final determination amount and a final determination date. What do we do in those circumstances? Here's an example API call: https://api.open.fec.gov/v1/legal/docs/admin_fines/1900?api_key=DEMO_KEY 

![Screen Shot 2021-05-05 at 3 39 41 PM](https://user-images.githubusercontent.com/12799132/117199483-3fbef900-adb8-11eb-8f66-ac12cd6417bf.png)

**Comment by JonellaCulmer** [comment]
> if there's no referral date and no referral amount, then we can eliminate the treasury referral row correct?

> if there's no challenge received date, then can we eliminate that row as well?

@patphongs Yes, I believe that's the case for both. 


> I also just found that sometimes there's no text for the challenge outcome even though there's a final determination amount and a final determination date. What do we do in those circumstances?

I don't think I quite understand. I don't see a challenge date in the API call. Is that "None" for a final determination date and the second $6,500 and 03/04/2009 are also for the final determination? Are you thinking we create language in the absence of a challenge date? I didn't think we were including the final determination in the table.

**Comment by patphongs** [comment]
> I don't think I quite understand. I don't see a challenge date in the API call. Is that "None" for a final determination date and the second $6,500 and 03/04/2009 are also for the final determination? Are you thinking we create language in the absence of a challenge date? I didn't think we were including the final determination in the table.

I have a loop in the middle of the table that populates the rows based on the number of documents in the call (see picture below). The initial column in that loop will populate the `challenge_outcome`. The challenge outcome will usually state something about the RTB or some other outcome. I'm starting to think that maybe it doesn't need to be a loop? Maybe we only want it to always be just one row, like we have it for the "Initial finding and proposed fine" and "Challenge received" rows. If that's the case, I can remove the loop, but this doesn't solve the problem where sometimes the `challenge_outcome` will be null. So we need to solve for what happens when it is null.

![Screen Shot 2021-05-06 at 9 38 13 AM](https://user-images.githubusercontent.com/12799132/117308684-c83ca880-ae4f-11eb-8fd0-3a68b92b54f2.png)

**Comment by patphongs** [comment]
> Yes, I believe that's the case for both.

Added additional logic to remove the challenge received row if there's no date and remove the treasury referral row if there's no treasury referral date and no treasury referral amount.

**Comment by JonellaCulmer** [comment]
@patphongs We may have an issue with the arrangement of the disposition. The challenge_outcome should probably follow challenge received, right?

And we're basically just populating the language for all the other dates and figures, right?

**Comment by patphongs** [comment]
Closing the loop on the challenge_outcome, final_determination_amount, and final_determination_date row. After discussing with @JonellaCulmer, we've decided that it does not need to be a loop for those rows, we only need to display one row. In addition, we are asking for feedback about what to display when the `challenge_outcome` is `null`, since presumably the case was not challenge and the fine was probably paid in full based on `check_amount`.
